### PR TITLE
Misc. fixes

### PR DESCRIPTION
**Pull request by snazy**
* make utility classes final w/ private c'tor
* enum using `==` instead of `.equals()`
* `String.isEmpty()` instead of `"".equals()`
* `HttpRequest` w/ `queryParam()` taking an `Integer`
* non-constant string in call to `printf` in `ReadReferences`
* collection field visibility in `RocksDbInstance`
* precompiled `Pattern`

### CODE REVIEW

File: clients/iceberg-views/src/main/java/org/apache/iceberg/view/VersionParser.java [hunk 1]
```diff
@@ -106,14 +106,19 @@ static Version fromJson(JsonNode node) {
...
+            break;
+          case CommonViewConstants.GENIE_ID:
+            builder.put(CommonViewConstants.GENIE_ID, JsonUtil.getString(field, summaryNode));
+            break;
+          case CommonViewConstants.ENGINE_VERSION:
+            builder.put(CommonViewConstants.ENGINE_VERSION, JsonUtil.getString(field, summaryNode));
+            break;
+          default:
+            builder.put(field, JsonUtil.getString(field, summaryNode));
+            break;
```
Comment (XN137) [inline clients/iceberg-views/src/main/java/org/apache/iceberg/view/VersionParser.java:121]: arent all switch cases doing the exact same thing? :thinking: 
```
builder.put(field, JsonUtil.getString(field, summaryNode));
```

**Review by XN137 — DISMISSED** [review]
typo in commit message: quaryParam

lgtm with 2 comments worth looking at

File: clients/iceberg-views/src/main/java/org/apache/iceberg/view/VersionParser.java (hunk 1 above)
Comment (snazy) [reply clients/iceberg-views/src/main/java/org/apache/iceberg/view/VersionParser.java:121]: Duh - did not want to touch iceberg-views at all.
Reverted this change.

@nastra there are a couple more issues, some are actual issues. But iceberg-views is WIP stuff - so IDK what to do with the module.

**Review by XN137 — APPROVED** [review]
commit msg still has typo ;)

**Pull request merged**